auto_backup/model/backup_scheduler.py

```diff
@@ -0,0 +1,375 @@
+        help=(
+            "The password from the user where the SFTP connection "
+            "should be made with. This is the password from the user"
+            " on the external server."
+        )
+    )
+    daystokeepsftp = fields.Integer(
+        string='Remove SFTP after x days',
+        default=30,
+        help=(
+            "Choose after how many days the backup should be deleted "
+            "from the FTP server. For example:\nIf you fill in 5 the "
+            "backups will be removed after 5 days from the FTP server."
+        )
+    )
+    sendmailsftpfail = fields.Boolean(
+        string='Auto. E-mail on backup fail',
+        help=(
+            "If you check this option you can choose to automaticly"
+            " get e-mailed when the backup to the external server failed."
+        )
+    )
+    emailtonotify = fields.Char(
+        string='E-mail to notify',
+        help=(
+            "Fill in the e-mail where you want to be"
+            " notified that the backup failed on the FTP."
+        )
+    )
+    lasterrorlog = fields.Text(
+        string='E-mail to notify',
+        help=(
+            "Fill in the e-mail where you want to be"
+            " notified that the backup failed on the FTP."
+        )
+    )
+
+    @api.multi
+    def _check_db_exist(self):
+        for rec in self:
+            db_list = db.exp_list(True)
+            if rec.name in db_list:
+                return True
+            return False
+
+    _constraints = [
+        (
+            _check_db_exist,
+            _('Error ,No such database exists'), ['name']
+        )
+    ]
+
+    @api.multi
+    def test_sftp_connection(self):
+        confs = self.search([])
+        # Check if there is a success or fail and write messages
+        messageTitle = ""
+        messageContent = ""
+        for rec in confs:
+            try:
+                conn_success = True
+                ipHost = rec.sftpip
+                portHost = rec.sftpport
+                usernameLogin = rec.sftpusername
+                passwordLogin = rec.sftppassword
+                # Connect with external server over SFTP, so we know sure that
+                # everything works.
+                srv = pysftp.Connection(host=ipHost, username=usernameLogin,
+                                        password=passwordLogin, port=portHost)
+                srv.close()
+                # We have a success.
+                messageTitle = _("Connection Test Succeeded!")
+                messageContent = _(
+                    "Everything seems properly set up for FTP back-ups!")
+            except Exception as e:
+                conn_success = False
+                messageTitle = _("Connection Test Failed!")
+                if len(rec.sftpip) < 8:
+                    messageContent += _(
+                        "\nYour IP address seems to be too short.\n")
+                messageContent += _("Here is what we got instead:\n")
+        if not conn_success:
```

Comment:
not sure about the variable scope. what if confs is empty? conn_success is never declared AFAICS